include/proto.h

```diff
@@ -364,6 +364,7 @@ enum {
 #define OF_APPLYABLE     0x00001000    // Set if object can be applied to another object
 #define OF_SAFETY        0x00002000    // Set if player has safety on (self only)
 #define OF_PLAYER_MASK   0x0001C000    // Mask to get player flag bits
+#define OF_LATENCY       0x0002C000    // Set if object contains latency information
```

Comment:
This is not how flags work.  Each flag must be a single bit that does not overlap with any others.  This value overlaps with the player flags, which could cause unforeseen problems if an object sets a flag and the client misinterprets it as a result.  The multi-bit values like OF_PLAYER_MASK are masks used to extract only those flags in a particular group (in this case, player-specific flags).

There are only 32 bits available (possibly only 28, would have to do a detailed server-side analysis), and I don't think we want to "spend" one on the single-purpose latency cube.

It also looks weird that this value is defined in the protocol, but never used on the server.  That's a sign that this isn't the right place for it.

I'll hold off reviewing the rest of this PR as I suspect this will require substantial changes.